packages/micropip/src/micropip/_micropip.py

```diff
@@ -116,9 +125,10 @@ def validate(self):
             # No checksums available, e.g. because installing
             # from a different location than PyPI.
             return
+        if not isinstance(self.data, BytesIO):
```

Comment:
Similarly, probably better to be explicit that we don't want to check the checksum if `self.parsed_url.scheme == "emfs":` rather than if it's not BytesIO which is an implementation detail.

Actually, is this necessary? It shouldn't have a digest with `emfs:`  URL and should return already above, no?

Reply:
It's needed by mypy.